evalml/data_checks/data_check_action_option.py

```diff
@@ -0,0 +1,129 @@
+            "code": self.action_code.name,
+            "data_check_name": self.data_check_name,
+            "metadata": self.metadata,
+            "parameters": self.parameters,
+        }
+        return action_option_dict
+
+    @staticmethod
+    def convert_dict_to_action(action_dict):
+        """Convert a dictionary into a DataCheckActionOption.
+
+        Args:
+            action_dict: Dictionary to convert into action. Should have keys "code", "data_check_name", and "metadata".
+
+        Raises:
+            ValueError: If input dictionary does not have keys `code` and `metadata` and if the `metadata` dictionary does not have keys `columns` and `rows`.
+
+        Returns:
+            DataCheckActionOption object from the input dictionary.
+        """
+        if "code" not in action_dict or "metadata" not in action_dict:
+            raise ValueError(
+                "The input dictionary should have the keys `code` and `metadata`."
+            )
+        if (
+            "columns" not in action_dict["metadata"]
+            or "rows" not in action_dict["metadata"]
+        ):
+            raise ValueError(
+                "The metadata dictionary should have the keys `columns` and `rows`. Set to None if not using."
+            )
+
+        return DataCheckActionOption(
+            action_code=DataCheckActionCode._all_values[action_dict["code"]],
+            metadata=action_dict["metadata"],
+            data_check_name=action_dict["data_check_name"]
+            if "data_check_name" in action_dict
+            else None,
+            parameters=action_dict["parameters"]
+            if "parameters" in action_dict
+            else None,
+        )
+
+    def _validate_parameters(self):
+        """Validate parameters associated with the action option."""
+        if self.parameters is None:
+            return
+        for _, parameter_value in self.parameters.items():
+            if "parameter_type" not in parameter_value:
+                raise ValueError("Each parameter must have a parameter_type key.")
+            if parameter_value["parameter_type"] == "global":
+                if "type" not in parameter_value:
+                    raise ValueError("Each global parameter must have a type key.")
+            elif parameter_value["parameter_type"] == "column":
+                if "columns" not in parameter_value:
+                    raise ValueError(
+                        "Each `column` parameter type must also have a `columns` key indicating which columns the parameter should address."
+                    )
+                columns = parameter_value["columns"]
+                if not isinstance(columns, dict):
+                    raise ValueError(
+                        "`columns` must be a dictionary, where each key is the name of a column and the associated value is a dictionary of parameters for that column."
+                    )
+                for _, column_parameters in columns.items():
```

Comment:
nit for this and the following for loop, since you don't use the first element of the item, can you instead iterate through `.values()` instead of `.items()`?